src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -404,6 +416,8 @@ public ObjectInstance run() {
         private int maxQueueSize = Integer.MAX_VALUE;
         private boolean registerMBean = REGISTER_MBEAN;
         private String mBeanName;
+        private Supplier<Object> contextSupplier = () -> null;
+        private Consumer<Object> contextConsumer = obj -> {};
```

Comment:
I'd be keen to use this to support tracing without additional ExecutorService wrapping or stack frames.

Thoughts on combining these into a single interface with a generic type?
e.g.
```java
public interface ExecutorContextHandler<T> {
    T createFromSubmittingThread();
    void applyContextToExecutorThread(T context);
    // perhaps this could also use some way to clear the context after a task completes?
    void clearContext();
}
```
My example method names are terrible, but I think this captures the idea.

Reply:
There isn't a convenient way (API-wise) to genericize these, hence going with Object.  I went with the simple functional objects so that they could be configured with static method refs, but I'm not married to it (maybe @kenfinnigan would like to offer an opinion as I wrote this to help him solve a Vert.x issue).  Maybe it's worth having an explicit clear as well; it depends on what the context is I suppose, and how detrimental it would be to leak it.

Reply:
I'm currently playing with this and may need to "tweak" this PR based on what I find.

Once I have something doing what is needed, we can then revisit better approaches and alternatives